### PR TITLE
Email Editor: Fix button rendering in Outlook [MAILPOET-5836]

### PR DESCRIPTION
**Pull request by mikejolley**
## Description

Fixes button rendering in outlook and across other email clients using inline styles (globally and from block settings) using table based markup.

I've included a new `AbstractBlockRenderer` class to house shared logic across renderers, and I've simplified the render logic as much as I think possible.

## Code review notes

- `safe_style_css` lets us define some non-standard properties so WordPress will still compile them.
- `AbstractBlockRenderer` has some utility for getting block styles and compiling CSS to save repetition.
- The original class had a `getMarkup` method with placeholders, but it seemed to make more sense to inline this using `sprintf`.
- I attempted using the conditional padding method from https://www.litmus.com/blog/a-guide-to-bulletproof-buttons-in-email-design before going back to tables which worked across most clients except the newer Outlook versions where no padding was present at all. Using a table for all seemed like the simplest approach given the requirements.
- The previous code had `mso-padding-alt` usage to allow the links to have padding in modern clients but the wrapper to have padding in outlook. This did not work on latest Outlook 365 and resulted in no padding being applied at all. Because of this I've opted to use padding only on the wrapper. cc @costasovo 
- I removed the border fallbacks which didn't appear to be needed. Border style solid only gets forced when borders are enabled, otherwise no border styles are added. I didn't notice any problems when testing email clients with this approach. Test coverage was updated to reflect this change.

I tested the rendered output using Email On Acid. Some examples:

o365_w10_lm_dt
gmailw10_chr26_win

Full report:

[Button testing_jHf8MtQzvSnZ5nSpA2owmXns5w06nWwwLl0rq6kP2k8Rh_2024-03-20 07.21.14.AM.zip](https://github.com/mailpoet/mailpoet/files/14667892/Button.testing_jHf8MtQzvSnZ5nSpA2owmXns5w06nWwwLl0rq6kP2k8Rh_2024-03-20.07.21.14.AM.zip)

## QA notes

Go to MailPoet > Emails > New Email > Create using new editor.

Test adding buttons with various colors, background colors, padding, and typography options. You can check the rendering by previewing the email.

To test other clients you can use Email on Acid as I did above.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5836](https://mailpoet.atlassian.net/browse/MAILPOET-5836)

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5836]: https://mailpoet.atlassian.net/browse/MAILPOET-5836?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lysyjan** [comment]
@mikejolley There are some conflicts and incorrect whitespaces. After merging your PHPCS improvement, it should be easy to fix. Can you please rebase it and fix it?
Thank you

File: mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Button.php
Comment (lysyjan) [inline mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Button.php]: I think this line is redundant because `AbstractBlockRenderer` is from the same namespace. The situation is similar in the `Column` and `Columns` renderer.

File: mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/AbstractBlockRenderer.php
Comment (lysyjan) [inline mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/AbstractBlockRenderer.php:19]: This method originally returned an object in the Column and Columns classes. Why did you decide to don't convert it again and use an array instead? I'm just curious.

**Review by lysyjan — DISMISSED** [review]
Good job @mikejolley. 
I have one question and noticed a detail, but I think we can move this PR forward.

**Comment by lysyjan** [comment]
@mikejolley Do you want to ask QA for testing, or can we merge it? In my opinion, we can merge it because you used Email on Acid here.

File: mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/AbstractBlockRenderer.php
Comment (mikejolley) [reply mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/AbstractBlockRenderer.php:19]: @lysyjan I think it was PHPStan, pre-commit hook, not liking the usage of stdClass. Rather than exclude everything it seemed easier to just leave in the original array format instead of casting.

File: mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Button.php
Comment (mikejolley) [reply mailpoet/lib/EmailEditor/Integrations/Core/Renderer/Blocks/Button.php]: Resolved in https://github.com/mailpoet/mailpoet/pull/5511/commits/1104d196af9de50211c044f3b09a538d427377ff

**Comment by mikejolley** [comment]
@lysyjan sounds good. I'll wait for CI to pass again then do it.

**Comment by mikejolley** [comment]
@lysyjan Commit dismissed review so just need ✅

**Review by lysyjan — APPROVED** [review]

**Pull request merged**
